database/src/main/resources/changesets/restart_and_recover_migration.xml

```diff
@@ -0,0 +1,138 @@
+                SELECT 1 FROM EXECUTION e2 WHERE
+                e2.WORKFLOW_EXECUTION_ID = e.WORKFLOW_EXECUTION_ID AND
+                e2.CALL_FQN = e.CALL_FQN AND
+                e2.IDX != -1)) AND
+              NOT EXISTS (                   -- filter out earlier attempts
+                SELECT 1 FROM EXECUTION e3 WHERE
+                e3.WORKFLOW_EXECUTION_ID = e.WORKFLOW_EXECUTION_ID AND
+                e3.CALL_FQN = e.CALL_FQN AND
+                e3.IDX = e.IDX AND
+                e3.ATTEMPT > e.ATTEMPT)
+        </sql>
+    </changeSet>
+
+    <changeSet id="JOB_STORE_RESULT_SIMPLETON_MIGRATION" author="mcovarr" >
+        <comment>
+            Restart/recover migration from 0.19 to 0.21.
+        </comment>
+        <customChange class="cromwell.database.migration.restart.table.JobStoreSimpletonMigration" />
+    </changeSet>
+
+    <changeSet id="BACKEND_KV_STORE_MIGRATION" author="mcovarr">
+        <comment>
+            Restart/recover migration from 0.19 to 0.21.
+        </comment>
+        <sql>
+            INSERT INTO BACKEND_KV_STORE(
+              WORKFLOW_EXECUTION_UUID,      -- 1
+              CALL_FQN,                     -- 2
+              JOB_SCATTER_INDEX,            -- 3
+              JOB_RETRY_ATTEMPT,            -- 4
+              STORE_KEY,                    -- 5
+              STORE_VALUE                   -- 6
+            )
+            SELECT
+              we.WORKFLOW_EXECUTION_UUID,   -- 1
+              e.CALL_FQN,                   -- 2
+              e.IDX,                        -- 3
+              e.ATTEMPT,                    -- 4
+              'jobId',                      -- 5
+              ei.INFO_VALUE                 -- 6
+            FROM
+              EXECUTION_INFO ei
+              JOIN EXECUTION e ON e.EXECUTION_ID = ei.EXECUTION_ID
+              JOIN WORKFLOW_EXECUTION we ON we.WORKFLOW_EXECUTION_ID = e.WORKFLOW_EXECUTION_ID
+            WHERE
+              ei.INFO_KEY = 'JES_RUN_ID' AND
+              ei.INFO_VALUE IS NOT NULL
+        </sql>
+    </changeSet>
+
+    <changeSet id="WORKFLOW_OPTIONS" author="mcovarr">
+        <comment>
+            Restart/recover migration from 0.19 to 0.21.
+        </comment>
+        <sql>
+            UPDATE WORKFLOW_STORE SET WORKFLOW_OPTIONS =
+              REPLACE(WORKFLOW_OPTIONS, 'defaultRuntimeOptions', 'default_runtime_attributes');
+
+            UPDATE WORKFLOW_STORE SET WORKFLOW_OPTIONS =
+              REPLACE(WORKFLOW_OPTIONS, 'workflowFailureMode', 'workflow_failure_mode');
+
+            UPDATE WORKFLOW_STORE SET WORKFLOW_OPTIONS =
+              REPLACE(WORKFLOW_OPTIONS, 'workflow_log_dir', 'final_workflow_log_dir');
+
+            UPDATE WORKFLOW_STORE SET WORKFLOW_OPTIONS =
+              REPLACE(WORKFLOW_OPTIONS, 'outputs_path', 'final_workflow_outputs_dir');
+
+            UPDATE WORKFLOW_STORE SET WORKFLOW_OPTIONS =
+              REPLACE(WORKFLOW_OPTIONS, 'call_logs_dir', 'final_call_logs_dir');
```

Comment:
Yeah this was a last minute hack.  I had been iteratively fighting my way through the preceding migrations for quite a while and was taken by surprise when they all worked and Cromwell actually began to restart workflows.  😄   The restarts failed for the mismatch in`defaultRuntimeOptions` vs `default_runtime_attributes` and this was a quick bandaid to get past that.

Anyhoo, I agree this is bad and should be fixed.  I don't see a way of doing this generally in straight SQL, but fortunately the scale is small enough that this could be handled as a custom task change class that actually parses the JSON map and only renames the keys.